commit fbd30b064f8f9607d500437b6fedc41431fd6cdc
Author: Even Rouault <even.rouault@spatialys.com>
Date:   Tue Dec 1 19:51:35 2020 +0100

    opj_t2_encode_packet(): avoid out of bound access of #1294, but likely not the proper fix

diff --git a/src/lib/openjp2/t2.c b/src/lib/openjp2/t2.c
index 71472b49..e452edd1 100644
--- a/src/lib/openjp2/t2.c
+++ b/src/lib/openjp2/t2.c
@@ -739,6 +739,15 @@ static OPJ_BOOL opj_t2_encode_packet(OPJ_UINT32 tileno,
                 continue;
             }
 
+            /* Avoid out of bounds access of https://github.com/uclouvain/openjpeg/issues/1294 */
+            /* but likely not a proper fix. */
+            if (precno >= res->pw * res->ph) {
+                opj_event_msg(p_manager, EVT_ERROR,
+                              "opj_t2_encode_packet(): accessing precno=%u >= %u\n",
+                              precno, res->pw * res->ph);
+                return OPJ_FALSE;
+            }
+
             prc = &band->precincts[precno];
             opj_tgt_reset(prc->incltree);
             opj_tgt_reset(prc->imsbtree);
